Neos.Media/Classes/Domain/Model/Adjustment/ResizeImageAdjustment.php

```diff
@@ -232,7 +235,7 @@ public function getMinimumWidth()
      * @return void
      * @api
      */
-    public function setRatioMode($ratioMode)
+    public function setRatioMode(string $ratioMode): void
```

Comment:
@kdambekalns this change is dangerous to be hidden inside a "cleanup task": it broke our site, because we were not even setting this property (thus it was "null" as there is no declared default in the property list), and then later `getRatioMode` was not returning a string anymore.
So this cleanup should have been paired with setting a proper default for `protected $ratioMode`. What do you think?

Reply:
Huh, sorry for that. I just checked the code again, and inside `ResizeImageAdjustment` the getter is never used, instead the property is read. And a check assigns a default of `ImageInterface::RATIOMODE_INSET` if no ratio mode is set. So it makes sense to assign a default to the class property…

Can you tell me where your code actually called `getRatioMode()`? Was that custom code or some "official" package?

Reply:
We generated Variants in the "code" (while importing images). Maybe it was not meant to be done like that, but "it worked", so we did:
```
        $resize = new ResizeImageAdjustment();
        $resize->setMaximumWidth($maximumWidth);
        $resize->setMaximumHeight($maximumHeight);
        $variant->addAdjustment($resize);
        $image->addVariant($variant);
```
This generated persisted records (in `neos_media_domain_model_adjustment_abstractimageadjustment`) which had a "null" value in the "ratioMode", which then exploded when it was materialized again as an object, because the setter could not be called.

Reply:
BTW all other persisted `AbstractImageAdjustment` that Neos generated have an empty string stored in this property (which is then a string, and does not crash). Which is kind of ugly if the setter explicitly states that one of the two constants are to be used.

Reply:
Ah, no, that's fine, usage-wise. Probably noone came across that error, since the thumbnails generated by Flow always have a ratio mode set.

But in your case unserializing from the DB actually breaks, since the setter does not accept `null`, right? So even if the getter would work now, stored instances would still be broken, if my brain parses PHP correcty…

Reply:
Yes, that is the problem, and we actually had to create a migration to fix the entries in the database. Since we might not be the only ones with that problem, such a migration could be useful for this kind of patch upgrade.

Maybe you could also pass that info to other core reviewers, to have an open eye when changing the strictness in setters/getters of domain models which might be already persisted with `null` values to provide a migration along with that change.

Reply:
The "default" of empty string makes sense, somehow. Well, not really, but it does no harm. It's a MySQL thing, not having `NULL` here even though it's allowed… and the column exists on all entries, even though all adjustments go into that table.

Reply:
See PR…